subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/RepoScriptBlockUtil.groovy

```diff
@@ -191,6 +213,13 @@ class RepoScriptBlockUtil {
                 void mirror(IvyArtifactRepository repo) {
                     ${mirrorConditions}
                 }
+                
+                // We see them as equal:
+                // https://repo.maven.apache.org/maven2/ and http://repo.maven.apache.org/maven2
+                String normalizeUrl(Object url) {
+                    String result = url.toString().replace('https://', 'http://')
```

Comment:
`new URL(url.toString()).getHost()` would also probably work. That is much more expensive, so I guess we can leave this as is. WDYT?

Reply:
I guess we can't - for example, the following repos are clearly not equal:

```
maven(url = "https://repo.gradle.org/gradle/libs-releases")
maven(url = "https://repo.gradle.org/gradle/libs-snapshots")
```